src/utils/decrypt/index.js

```diff
@@ -5,32 +5,21 @@ module.exports = decrypt;
 
 // https://tools.ietf.org/html/draft-ietf-webpush-encryption-03
 function decrypt(object, keys) {
-  try {
-    const cryptoKey = object.appData
-      .find(item => item.key === 'crypto-key')
-      .value.slice(3);
-    const salt = object.appData
-      .find(item => item.key === 'encryption')
-      .value.slice(5);
-    const dh = crypto.createECDH('prime256v1');
-    dh.setPrivateKey(keys.privateKey, 'base64');
-    const params = {
-      version    : 'aesgcm',
-      authSecret : keys.authSecret,
-      dh         : cryptoKey,
-      privateKey : dh,
-      salt,
-    };
-    try {
-      const decrypted = ece.decrypt(object.rawData, params);
-      return JSON.parse(decrypted);
-    } catch (e) {
-      console.error('Error while decrypting incoming message');
-      console.error(e);
-      return;
-    }
-  } catch (e) {
-    // Not a notification
-    return;
-  }
+  const cryptoKey = object.appData
+    .find(item => item.key === 'crypto-key')
+    .value.slice(3);
+  const salt = object.appData
+    .find(item => item.key === 'encryption')
+    .value.slice(5);
+  const dh = crypto.createECDH('prime256v1');
+  dh.setPrivateKey(keys.privateKey, 'base64');
+  const params = {
+    version    : 'aesgcm',
+    authSecret : keys.authSecret,
+    dh         : cryptoKey,
+    privateKey : dh,
+    salt,
+  };
+  const decrypted = ece.decrypt(object.rawData, params);
+  return JSON.parse(decrypted);
```

Comment:
Why are you removing the `try`/`catch` stuff? Does anything ensure that it won't fail now?

Reply:
I think these were mostly masking errors. When I originally filed https://github.com/MatthieuLemoine/push-receiver/issues/10 the notifications were silently failing with "not a notification". Now that the code above handles all the different types (i.e. everything in `buildProtobufFromTag`) I don't think this should happen. I'd rather it throw errors and be noisy :)